flocker/common/test/test_interface.py

```diff
@@ -132,3 +133,54 @@ def test_raise(self, logger):
         logging_dummy = LoggingDummy(Dummy(result))
         exception = self.assertRaises(ValueError, logging_dummy.raise_method)
         self.assertIs(result, exception)
+
+
+class InterfaceHolder(PClass):
+    """
+    ``PClass`` that has a field with a ``provides`` invariant.
+    """
+    value = field(invariant=provides(IDummy), mandatory=True)
+
+
+class ProvidesTests(SynchronousTestCase):
+    """
+    Tests for ``provides``.
+    """
+
+    def test_interface_accepted(self):
+        """
+        When an object providing the given interface is provided,
+        no invariant exception is raised.
+        """
+        InterfaceHolder(value=Dummy(object()))
+
+    def test_other_rejected(self):
+        """
+        When an object not providing the given interface is provided,
+        an ``InvariantException`` is raised.
+        """
+        self.assertRaises(
+            InvariantException,
+            InterfaceHolder, value=object(),
+        )
+
+    def test_invariant_message(self):
+        """
+        When an object not providing the given interface is provided,
+        an ``InvariantException`` is raised.
+        """
+        exception = self.assertRaises(
+            InvariantException,
+            InterfaceHolder, value=True,
+        )
+        self.assertEqual(
+            exception.invariant_errors,
+            ("True doesn't provide IDummy",),
+        )
+
+    def test_invariant_name(self):
```

Comment:
:grey_question: Do we require a docstring on all test methods? it looks like we do, but I'm not sure what it adds in this case. Name is rather descriptive, test is simple... Fix if you desire to and just missed this one, I'm indifferent.